commit d0eb22f3ba557b9e98836995c813ea77c5e7c2a5
Author: Sage Weil <sweil@redhat.com>
Date:   Wed Jul 31 04:14:01 2019 -0500

    mon/health_checks: associate a count with health_alert_t
    
    0 means this is a singleton.  Otherwise, we can sum this up, either
    via merge() or get_or_add().
    
    We always structure this so the count goes toward zero (more healthy), so
    if a value is too low, then we count how much too low it is.
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/PyModuleRegistry.cc b/src/mgr/PyModuleRegistry.cc
index 16c43e40e07..ac92c9282cc 100644
--- a/src/mgr/PyModuleRegistry.cc
+++ b/src/mgr/PyModuleRegistry.cc
@@ -386,7 +386,8 @@ void PyModuleRegistry::get_health_checks(health_check_map_t *checks)
         ss << dependency_modules.size()
 	   << " mgr modules have failed dependencies";
       }
-      auto& d = checks->add("MGR_MODULE_DEPENDENCY", HEALTH_WARN, ss.str());
+      auto& d = checks->add("MGR_MODULE_DEPENDENCY", HEALTH_WARN, ss.str(),
+			    dependency_modules.size());
       for (auto& i : dependency_modules) {
 	std::ostringstream ss;
         ss << "Module '" << i.first << "' has failed dependency: " << i.second;
@@ -402,7 +403,8 @@ void PyModuleRegistry::get_health_checks(health_check_map_t *checks)
       } else if (failed_modules.size() > 1) {
         ss << failed_modules.size() << " mgr modules have failed";
       }
-      auto& d = checks->add("MGR_MODULE_ERROR", HEALTH_ERR, ss.str());
+      auto& d = checks->add("MGR_MODULE_ERROR", HEALTH_ERR, ss.str(),
+			    failed_modules.size());
       for (auto& i : failed_modules) {
 	std::ostringstream ss;
         ss << "Module '" << i.first << "' has failed: " << i.second;
